src/openforms/formio/components/custom.py

```diff
@@ -770,6 +771,156 @@ def as_json_schema(component: Component) -> JSONObject:
+                {
+                    key: (
+                        datetime.strptime(value, "%Y-%m-%d").date()
+                        if key == "dateOfBirth" and value
+                        else value
+                    )
+                    for key, value in child.items()
+                    if key
+                    not in (
+                        "dateOfBirthPrecision",
+                        "lastName",
+                        "affixes",
+                        "initials",
+                    )
+                }
+                for child in prefill_data[fm_immutable_variable.key]
+            ]
+
+            if not self.validate_selected_children(children):
+                raise serializers.ValidationError(_("Invalid data."))
+
+            # We also receive the boolean `selected` which is set on the frontend and
+            # therefore should not be part of the data for validation (in combination
+            # with the retrieved data from the external source)
+            for child in children:
+                child.pop("selected", None)
+
+            if initial_value and initial_value != children:
+                raise serializers.ValidationError(
+                    _("The family members prefill data may not be altered.")
+                )
+
+    # TODO
+    # Add proper type hints when #2324 is completed (sync the frontend with the backend)
+    def validate_selected_children(self, children) -> bool:
+        """
+        Validate the values of the `selected` key defined on the frontend.
+
+        When the component allows selecting children, the value of the `selected` key
+        should be of type boolean, otherwise it should be None.
+        """
+        selection_allowed = self.component["enableSelection"]
+        if selection_allowed:
+            return all(isinstance(child["selected"], bool) for child in children)
+        else:
+            return all(child["selected"] is None for child in children)
+
+
+@register("children")
+class Children(BasePlugin[ChildrenComponent]):
+    formatter = DefaultFormatter
+
+    def build_serializer_field(self, component: ChildrenComponent) -> ChildListField:
+        return ChildListField(component=component)
+
+    @staticmethod
+    def as_json_schema(component: ChildrenComponent) -> JSONObject:
+        label = component.get("label", "Children")
+        schema = {
+            "title": label,
+            "type": "array",
+            "items": {
+                "type": "object",
+                "required": ["bsn"],
+                "properties": {
+                    "bsn": {
+                        "type": "string",
+                        "pattern": r"^\d{9}$",
+                        "format": "nl-bsn",
+                    },
+                    "firstNames": {"type": "string"},
+                    "dateOfBirth": {"type": "string", "format": "date"},
+                },
+                "additionalProperties": False,
```

Comment:
is that true? will we not send extra properties like `affixes` and `lastName` etc.?